python/rmm/rmm/_lib/helper.pyx

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helper functions for rmm"""
+
+import re
+
+from libc.stdlib cimport atof
+
+
+cdef dict BYTE_SIZES = {
+    'b': 1,
+    '': 1,
+    'kb': 1000,
+    'mb': 1000000,
+    'gb': 1000000000,
+    'tb': 1000000000000,
+    'pb': 1000000000000000,
+    'k': 1000,
+    'm': 1000000,
+    'g': 1000000000,
+    't': 1000000000000,
+    'p': 1000000000000000,
```

Comment:
```suggestion
```

These are not valid "byte-spec" suffixes by the regex we now have.

Reply:
I changed the regex slightly so the b/B is optional  @wence-